# HG changeset patch
# User Rik <rik@octave.org>
# Date 1461870741 25200
#      Thu Apr 28 12:12:21 2016 -0700
# Node ID 2e07dcdc613980000ed737ddad3b3e0b355126ed
# Parent  b4354327d2b51f8452db3933bbd75e60aa88a073
legend.m: Check handle is valid before delete (partial fix bug #47699).

* legend.m (deletelegend2): Check that invisible object t1 is valid
before trying to delete it.  It may already have been deleted.
* legend.m (hideshowlegend): Use variable name hlegend, not ca,
for truth in advertising.

diff --git a/scripts/plot/appearance/legend.m b/scripts/plot/appearance/legend.m
--- a/scripts/plot/appearance/legend.m
+++ b/scripts/plot/appearance/legend.m
@@ -1143,21 +1143,21 @@ function hideshowlegend (h, ~, ca, pos1,
         endif
       unwind_protect_cleanup
         set (ca(i), "units", units);
       end_unwind_protect
     endif
   endfor
 endfunction
 
-function deletelegend1 (h, ~, ca)
-  if (isaxes (ca)
+function deletelegend1 (h, ~, hlegend)
+  if (isaxes (hlegend)
       && (isempty (gcbf ()) || strcmp (get (gcbf (), "beingdeleted"), "off"))
-      && strcmp (get (ca, "beingdeleted"), "off"))
-    delete (ca);
+      && strcmp (get (hlegend, "beingdeleted"), "off"))
+    delete (hlegend);
   endif
 endfunction
 
 function deletelegend2 (h, ~, ca, pos, outpos, t1, hplots)
   for i = 1 : numel (ca)
     if (isaxes (ca(i))
         && (isempty (gcbf ()) || strcmp (get (gcbf (), "beingdeleted"), "off"))
         && strcmp (get (ca(i), "beingdeleted"), "off"))
@@ -1167,18 +1167,20 @@ function deletelegend2 (h, ~, ca, pos, o
           set (ca(i), "units", "points");
           set (ca(i), "position", pos, "deletefcn", "");
         unwind_protect_cleanup
           set (ca(i), "units", units);
         end_unwind_protect
       endif
     endif
   endfor
-  set (t1, "deletefcn", "");
-  delete (t1);
+  if (ishandle (t1))
+    set (t1, "deletefcn", "");
+    delete (t1);
+  endif
   for i = 1 : numel (hplots)
     if (ishandle (hplots(i)) && strcmp (get (hplots(i), "type"), "line"))
       dellistener (hplots(i), "color");
       dellistener (hplots(i), "linestyle");
       dellistener (hplots(i), "linewidth");
       dellistener (hplots(i), "marker");
       dellistener (hplots(i), "markeredgecolor");
       dellistener (hplots(i), "markerfacecolor");
